examples/tv-casting-app/tv-casting-common/support/ChipDeviceEventHandler.cpp

```diff
@@ -76,7 +76,12 @@ void ChipDeviceEventHandler::Handle(const chip::DeviceLayer::ChipDeviceEvent * e
             [](void * context, const chip::ScopedNodeId & peerId, CHIP_ERROR error) {
                 ChipLogError(AppServer, "ChipDeviceEventHandler::Handle: Connection to CastingPlayer failed");
                 CastingPlayer::GetTargetCastingPlayer()->mConnectionState = CASTING_PLAYER_NOT_CONNECTED;
-                support::CastingStore::GetInstance()->Delete(*CastingPlayer::GetTargetCastingPlayer());
+                CHIP_ERROR err = support::CastingStore::GetInstance()->Delete(*CastingPlayer::GetTargetCastingPlayer());
```

Comment:
Is this automatically deleting the casting player entry when connection fails? Would this prevent the customer from maintaining connections to players on 2 different networks (eg. primary home and secondary home)?

Reply:
It will not prevent a customer from maintaining connections to 2 different CastingPlayers (on different / same networks) because this only deletes the entry for one CastingPlayer from the CastingStore (the one we could not connect to). This is done to circumvent the cases where the nodeId/fabricIndex of a CastingPlayer is no longer valid, say in the case of a factory reset. 

Also note, the VerifyOrEstablishConnection API will run the full commissioning process on the next call anyway, so a  client should be able to connect even with a CastingPlayer whose nodeId/fabricIndex data became stale previously.